libcodechecker/analyze/plist_parser.py

```diff
@@ -411,84 +412,101 @@ def write(self, files, reports, analyzed_source_file, output=sys.stdout):
+                              "deduplication of an already processed report!")
+                    LOG.debug("Path hash: %s", path_hash)
+                    LOG.debug(report)
+                    continue
+
+                self._processed_path_hashes.add(path_hash)
+
+                events = [i for i in report.bug_path
+                          if i.get('kind') == 'event']
+                f_path = report.files[events[-1]['location']['file']]
+                if self.skiplist_handler and \
+                        self.skiplist_handler.should_skip(f_path):
+                    LOG.debug("Skipped report in '%s'", f_path)
+                    LOG.debug(report)
+                    continue
+
+                last_report_event = report.bug_path[-1]
+                source_file = \
+                    report.files[last_report_event['location']['file']]
+                report_line = last_report_event['location']['line']
+                report_hash = \
+                    report.main['issue_hash_content_of_line_in_context']
+                checker_name = report.main['check_name']
+
+                if skip_report(report_hash, source_file, report_line,
+                               checker_name, self.src_comment_handler):
+                    continue
+
+                file_stats[f_path] += 1
+                severity = self.__severity_map.get(checker_name,
+                                                   'UNSPECIFIED')
+                severity_stats[severity] += 1
+                report_count["report_count"] += 1
+
+                output.write(self.__format_bug_event(checker_name,
+                                                     severity,
+                                                     last_report_event,
+                                                     source_file))
+                output.write('\n')
+                output.write(self.__format_location(last_report_event,
+                                                    source_file))
+                output.write('\n')
+
+                if self.print_steps:
+                    output.write('  Report hash: ' + report_hash + '\n')
+                    output.write('  Steps:\n')
+                    for index, event in enumerate(events):
+                        output.write(index_format % (index + 1))
+                        source_file = report.files[event['location']['file']]
+                        output.write(self.__format_bug_event(None,
+                                                             None,
+                                                             event,
+                                                             source_file))
+                        output.write('\n')
+                output.write('\n')
+
+                non_suppressed += 1
+
+            base_file = os.path.basename(file_path)
+            if non_suppressed == 0:
+                output.write('Found no defects while analyzing %s\n' %
```

Comment:
Output should be changed to `Found defects in a file...` because it is possible that we print out a report in a header file which was found during the analysis of a cpp file.